src/unused_finder/unused_finder.rs

```diff
@@ -164,122 +125,155 @@ impl UnusedFinder {
+                .filter(|(_, graph_file)| !graph_file.is_test_file && !graph_file.is_used),
         );
-        let unused_files_items: HashMap<String, Vec<ExportedItemReport>> = self
-            .graph
+
+        // Set initial frontier as the list of test files
+        let mut frontier: Vec<String> = self
+            .src_files
+            .iter()
+            .filter_map(|walkfile| match walkfile.is_test_file {
+                true => Some(walkfile.source_file_path.to_string()),
+                false => None,
+            })
+            .collect();
+        // Do graph bfs for test files
+        for _ in 0..100_000 {
+            frontier = graph.bfs_step(frontier);
+        }
+        if !frontier.is_empty() {
+            return Err(napi::Error::new(
+                napi::Status::GenericFailure,
+                "exceeded max iterations".to_string(),
+            ));
+        }
+
+        // Get file paths only reachable by test files
+        let test_only_used_files: Vec<String> = graph
             .files
             .iter()
-            .filter_map(|(file_path, info)| {
-                if info.is_used {
-                    match file_path_exported_items_map.remove(file_path) {
-                        Some(mut exported_items) => {
-                            let unused_exports = &info.unused_exports;
-                            if unused_exports.is_empty() {
-                                return None;
-                            }
-                            let unused_exports = exported_items
-                                .drain(0..)
-                                .filter(|exported| {
-                                    unused_exports
-                                        .iter()
-                                        .any(|unused| unused.to_string() == exported.id.to_string())
-                                })
-                                .collect();
-                            return Some((file_path.to_string(), unused_exports));
-                        }
-                        None => return None,
+            .filter_map(|(file_path, graph_file)| {
+                // If after a graph bfs on test files is marked and used
+                // And file list of reachable files from entries contains said file
+                // said file is reachable only by test files
+                // If a file file was previously marked as unused, and graph_file.is_used == true after bfs on test files
+                if graph_file.is_used && !unused_prod_files.contains_key(file_path) {
+                    // If allow list contains a file or pattern that matches said file, we don't include it in the list
+                    if !allow_list.iter().any(|p| p.matches(&file_path)) {
+                        return Some(file_path.to_string());
                     }
                 }
                 None
             })
             .collect();
 
+        unused_items.iter_mut().for_each(|(file, items)| {
```

Comment:
I don't see a clear reason to segregate "files that are only reachable by tests" into its own category of reachable vs unreachable.

Reply:
If a file or item is only being imported by test files it could potentially be removed or moved under `**/test/**` if it doesn't serve another purpose. The intention with having two walks is to somehow be able to tell owa-extension to highlight a warning/info on this items as "Potentially used".